src/Microsoft.Dnx.Runtime/FrameworkReferenceResolver.cs

```diff
@@ -277,7 +342,7 @@ private static FrameworkInformation GetFrameworkInformation(DirectoryInfo direct
 
                             var entry = new AssemblyEntry();
                             entry.Version = version != null ? Version.Parse(version) : null;
-                            frameworkInfo.Assemblies.Add(assemblyName, entry);
+                            frameworkInfo.Assemblies[assemblyName] = entry;
```

Comment:
Looks like MSBuild lets you select both. In theory we could do that by allowing a version in the `frameworkAssemblies` entry but that's a larger change. I suggest that if we care about supporting that, we make a new bug for it. Until then, we default to the highest version.